client/js/jsonedit.js

```diff
@@ -34,6 +34,13 @@ const jeCommands = [
       return widgets.has(m[1]);
     }
   },
+  {
+    name: 'upload a different asset',
+    context: '"(/assets/[0-9_-]+)"',
```

Comment:
Not sure what you could do to improve this functionality, but if the user misses any character at the end of the asset URL the results after upload will be broken.

Also a way to add assets where none has been uploaded yet would be nice. (I can type a dummy asset URL to replace).

Even as it is now, this is a huge improvement to work in JSON with uploaded assets.